### PR TITLE
Add install/uninstall support

### PR DESCRIPTION
**Pull request by nocnokneo**
Including generation of `KalmanConfig.cmake` and `KalmanConfigVersion.cmake` files to support `find_package(Kalman)`.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -1,36 +1,42 @@
-cmake_minimum_required(VERSION 2.8)
-project(Kalman)
+cmake_minimum_required(VERSION 3.0)
```
Comment (mherb) [inline CMakeLists.txt]: I'd prefer retaining backwards-compatibility to 2.8 as it is still the default on (for instance) Ubuntu 14.04 LTS

File: cmake/KalmanConfig.cmake.in [hunk 2]
```diff
@@ -0,0 +1,21 @@
+#                                               -*- cmake -*-
+#
+#  KalmanConfig.cmake
+
+# Use the following variables to compile and link against Eigen:
```
Comment (mherb) [inline cmake/KalmanConfig.cmake.in]: Comments should probably be updates as we're not eigen ;)

File: CMakeLists.txt [hunk 3]
```diff
@@ -1,36 +1,42 @@
...
+project(Kalman VERSION 0.1.0 LANGUAGES CXX)
+
+include(CTest)
+
+set(KALMAN_DEFINITIONS "-std=c++11")
```
Comment (mherb) [inline CMakeLists.txt]: I wouldn't consider the used cpp standard a library definition, the application using the lib should decide which standard it wants to use (could be c++14 or later as well). Any specific reason why this is included here?

**Comment by mherb** [comment]
Looks very neat, thank you! Do you want to update the PR on the noted things before merging?

File: CMakeLists.txt (hunk 1 above)
Comment (nocnokneo) [inline CMakeLists.txt]: Makes sense. Fixed.

File: CMakeLists.txt (hunk 3 above)
Comment (nocnokneo) [inline CMakeLists.txt]: Ya, I agree. I was just using the naming convention I had seen in other *Config.cmake files, but I'vs also seen names like FOO_CXX_FLAGS so thats the style I've now changed it to.

**Pull request merged**